src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -0,0 +1,201 @@
+			//			_cacheFolder.Dispose();
+
+			GC.SuppressFinalize(this);
+		}
+
+		private void TryToDeleteCachedImages()
+		{
+			lock (this)
+			{
+				foreach(var path in _originalPathToProcessedVersionPath.Values)
+				{
+					try
+					{
+						if (File.Exists(path))
+						{
+							File.Delete(path);
+							Debug.WriteLine("RuntimeImageProcessor Successfully deleted: " + path);
+						}
+					}
+					catch (Exception e)
+					{
+						Debug.WriteLine("RuntimeImageProcessor Dispose(): " + e.Message);
+					}
+				}
+				_originalPathToProcessedVersionPath.Clear();
+			}
+		}
+
+		public string GetPathToResizedImage(string originalPath)
+		{
+			//don't mess with Bloom UI images
+			if (new[] {"/img/", "placeHolder", "Button"}.Any(s => originalPath.Contains(s)))
+				return originalPath;
+
+			lock (this)
+			{
+				string pathToProcessedVersion;
+				if (_originalPathToProcessedVersionPath.TryGetValue(originalPath, out pathToProcessedVersion))
+				{
+					if (File.Exists(pathToProcessedVersion) &&
+					    new FileInfo(originalPath).LastWriteTimeUtc <= new FileInfo(pathToProcessedVersion).LastWriteTimeUtc)
+					{
+						return pathToProcessedVersion;
+					}
+					else
+					{
+						_originalPathToProcessedVersionPath.Remove(originalPath);
+					}
+				}
+
+				using (var originalImage = PalasoImage.FromFile(originalPath))
+				{
+					//if it's a jpeg, we don't resize, we don't mess with transparency, nothing. These things
+					//are scary in .net. Just send the original back and wash our hands of it.
+					if (ImageUtils.AppearsToBeJpeg(originalImage))
+					{
+						return originalImage.OriginalFilePath;
+					}
+
+					double shrinkFactor = 1.0;
+
+#if ShrinkLargeImages // at the moment, we're suspecting that it may be better to let the browser do the shrinking
+
+					//if its a small image, like a creative commons logo, we don't try and resize it
+					if (originalImage.Image.Width > TargetDimension || originalImage.Image.Height > TargetDimension)
+					{
+						var maxDimension = Math.Max(originalImage.Image.Width, originalImage.Image.Height);
+						//enhance: if we had a way of knowing what the target dimension actually was, we'd use that, of course
+						shrinkFactor = (TargetDimension/(double) maxDimension);
+					}
+#endif
+					var destWidth = (int) (shrinkFactor*originalImage.Image.Width);
+					var destHeight = (int) (shrinkFactor*originalImage.Image.Height);
+
+					using (var processedBitmap = new Bitmap(destWidth, destHeight))
+					{
+						using (Graphics g = Graphics.FromImage((Image) processedBitmap))
+						{
+							//in version 1.0, we used .NearestNeighbor. But if there is a border line down the right size (as is common for thumbnails that,
+							//are, for example, re-inserted into Teacher's Guides), then the line gets cut off. So I switched it to HighQualityBicubic
+							g.InterpolationMode = InterpolationMode.HighQualityBicubic; //.NearestNeighbor;//or smooth it: HighQualityBicubic
```

Comment:
Does this make any difference when we're not changing the size? Any thought that there would be a better choice (equal quality, faster?) for the no-size-change situation?